### ISSUE TITLE
[BUG] [Linter] H025 false positive

### ISSUE THREAD

**Issue by heitorado** [Riverside-Healthcare/djLint#364]
- [x] I'm on the [latest version](https://pypi.org/project/djlint/) of djLint
- [x] I've searched the [issues](https://github.com/Riverside-Healthcare/djLint/issues)
- [x] I've read the [docs](https://djlint.com)

## System Info
 - OS: e.g. ubuntu 22.04
 - Python Version (``3.10.4``)
 - djLint Version (``1.12.1``)
 - template language: `django`

## Issue
When HTML tags are used inside strings on Django templates (e.g. as an attribute for a filter), H025 shows a false positive, thinking that the tag is orphan.

## How To Reproduce
These two code snippets fall into the H025 rule but none of them should.
`{% if tag|startswith:"<del>" %}`
`{% if tag|startswith:"<ins>" %}`

**Comment by christopherpickering** [comment]
@heitorado thanks 😀  this one is a very hard problem to solve. I wonder if the linter rules need another option to say if the matches are allowed to be inside template syntax or not. Some of the other rules will probably fall in the same boat.

like `{% if tag|startswith:"<img />" %}` triggers several other rules too.

I'm thinking something like this:

![image](https://user-images.githubusercontent.com/17788706/188875628-ddbebc39-63dc-4a44-8c3f-fa15a8fac6cd.png)

and then checking the conditions before reporting:

![image](https://user-images.githubusercontent.com/17788706/188875807-47adddce-e0fa-45b1-9f48-3f7f31174ef5.png)

**Comment by christopherpickering** [comment]
I'm thinking this is also the same as/similar to #357.

**Comment by silverwind** [comment]
I'm also seeing false-positive H25 (Tag seems to be an orphan) on this golang syntax:

```handlebars
{{$link := printf "<a href=\"%s\">%s</a>" $url $text}}
```

Apparently it's tripping on the escaped double quotes as this case is accepted:

```handlebars
{{$link := printf `<a href="%s">%s</a>` $url $text}}
```

**Comment by domdfcoding** [comment]
I'm seeing a mix of this and #91. I've got a comment 32 lines down from the reported error, `{# <div class="d-grid gap-1 d-md-flex justify-content-md-end h-100"> #}`. If I delete the comment it no longer complains that the earlier opening div is an orphan. 

If I follow through the HTML, on the assumption it thinks the commented out div is actually part of the markup, I can see why it thinks the earlier div is an orphan as there's an off by one for all subsequent closing divs. But the comment *shouldn't* be considered part of the markup in the first place.
